cloud_function/main.py

```diff
@@ -0,0 +1,127 @@
+        "pipeline": "AllOfUsArrays",
+        "project": "gotc-dev",
+        "items": [{}]
+    }
+    response = requests.post(
+        url=f"{wfl_url}/api/v1/exec",
+        headers=headers,
+        json=payload
+    )
+    return response.json().get('uuid')
+
+def update_workload(headers, workload_uuid, input_data):
+    input_data['uuid'] = workload_uuid
+    print(f"Updating workload {workload_uuid}")
+    response = requests.post(
+        url=f"{wfl_url}/api/v1/append_to_aou",
+        headers=headers,
+        json=input_data
+    )
+    workflows = response.json()
+    return [each['uuid'] for each in workflows]
+
+def submit_aou_workload(event, context):
+    """Background Cloud Function to be triggered by Cloud Storage.
+    Args:
+         event (dict):  The dictionary with data specific to this type of
+         event. The `data` field contains the PubsubMessage message. The
+         `attributes` field will contain custom attributes if there are any.
+         context (google.cloud.functions.Context): The Cloud Functions event
+         metadata. The `event_id` field contains the Pub/Sub message ID. The
+         `timestamp` field contains the publish time.
+    """
+
+    if default_credentials:
+        headers = get_auth_headers()
+        client = storage.Client()
+    else:
+        credentials = get_test_credentials()
+        headers = {
+            'Authorization': 'Bearer {}'.format(credentials.token)
+        }
+        client = storage.Client(credentials=credentials, project=project)
+
+    # Get sample manifest/metadata file
+    bucket = client.bucket(event['bucket'])
+    manifest_path = get_manifest_path(event['name'])
+    manifest_blob = bucket.blob(manifest_path)
+
+    try:
+        manifest_file_content = manifest_blob.download_as_string()
+    except exceptions.NotFound as e:
+        print(f'File not found: {manifest_path}')
+        return
+
+    # Parse manifest file to get input file paths
+    input_data = json.loads(manifest_file_content)
+    notification = input_data['notifications'][0]
+    input_values = notification.values()
+    input_files = set([f for f in input_values if str(f).startswith("gs://")])
+
+    # Check if the input files have been uploaded
+    upload_complete = True
+    for gs_url in input_files:
+        file_blob = storage.Blob.from_string(gs_url)
+        file_metadata = bucket.get_blob(file_blob.name)
+        if not file_metadata:
+            upload_complete = False
+            print(f"File not found: {gs_url}")
+            return
+
+    if upload_complete:
+        sample_alias = notification.get('sample_alias')
+        print(f"Sample upload completed for {sample_alias}")
+        cromwell_url = input_data.get('cromwell')
+        workload_uuid = get_or_create_workload(headers, cromwell_url)
+        workflow_ids = update_workload(headers, workload_uuid, input_data)
+        print(f"Started cromwell workflows: {workflow_ids}")
```

Comment:
I'm not sure about Cloud Functions logging features, but I can see we/hOps will be interested in being able to filter different information this cloud function logs, such as: what are the successfully submitted workflows (UUID), their chipwell-barcodes, which samples are throwing `File not found Erros`, etc. Not sure if we should do it here or as a follow up.

Reply:
That's a good point! Instead of `"Sample upload completed for {sample_alias}"` it seems like `"Sample upload completed for {chipwell_barcode}-{analysis_version}" would be more useful? Also we can improve this in a follow-up PR too based on feedback from hOps!